### PR TITLE
The big 1.17 cleanup - part one

### PR DESCRIPTION
**Pull request by IMS212**

### CODE REVIEW

**Review by coderbot16 — CHANGES_REQUESTED** [review]
This is a nice improvement. I have a few comments though.

File: src/main/java/net/coderbot/iris/mixin/MixinLevelRenderer.java [hunk 1]
```diff
@@ -138,63 +123,19 @@
...
-	}*/
-
-	// TODO(21w10a): Deal with render hooks
 	@Inject(method = RENDER, at = @At(value = "INVOKE", target = RENDER_SKY, shift = At.Shift.AFTER))
 	private void iris$endSky(PoseStack poseStack, float tickDelta, long limitTime, boolean renderBlockOutline, Camera camera, GameRenderer gameRenderer, LightTexture lightTexture, Matrix4f matrix4f, CallbackInfo callback) {
 		// After the sky has rendered, there isn't a clear phase.
 		pipeline.setPhase(WorldRenderingPhase.OTHER);
 	}
 
-	/*@Inject(method = "renderLevel", at = @At(value = "INVOKE", target = RENDER_CLOUDS))
```
[the remark this answers is not in the document]
Comment (IMS212) [reply src/main/java/net/coderbot/iris/mixin/MixinLevelRenderer.java:156]: Should we replace the TODO's with an actual phase?

[the remark this answers is not in the document]
Comment (coderbot16) [reply src/main/java/net/coderbot/iris/mixin/MixinLevelRenderer.java:156]: If you'd like & know how to implement it, sure.

File: src/main/java/net/coderbot/iris/gl/shader/ProgramCreator.java [hunk 2]
```diff
@@ -20,7 +20,6 @@ public static int create(String name, GlShader... shaders) {
 		GlStateManager._glBindAttribLocation(program, 11, "mc_midTexCoord");
 		GlStateManager._glBindAttribLocation(program, 12, "at_tangent");
 
-		// TODO(21w10a): more hardcoding!!!!
```
[the remark this answers is not in the document]
Comment (coderbot16) [reply src/main/java/net/coderbot/iris/gl/shader/ProgramCreator.java:23]: (ie, we can keep this TODO for now)

File: src/main/java/net/coderbot/iris/mixin/texunits/MixinBufferVertexConsumer.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
 	 * @reason FabricMC Mixin does not support injections into interfaces
 	 * @author coderbot16
 	 */
-	/*TODO(21w10a): Replace texunit hooks
+	/* Unused on 1.17
```
[the remark this answers is not in the document]
Comment (IMS212) [reply src/main/java/net/coderbot/iris/mixin/texunits/MixinBufferVertexConsumer.java:28]: We can delete the whole texunits package, is that ok?

[the remark this answers is not in the document]
Comment (coderbot16) [reply src/main/java/net/coderbot/iris/mixin/texunits/MixinBufferVertexConsumer.java:28]: It should be irrelevant on 1.17, yeah.

[the remark this answers is not in the document]
Comment (coderbot16) [reply src/main/java/net/coderbot/iris/mixin/texunits/MixinBufferVertexConsumer.java:28]: (And hopefully, at some point it'll be able to be removed from 1.16 too. It's not a great piece of code.)

**Comment by IMS212** [comment]
Fixed all reviews.

**Comment by coderbot16** [comment]
Looks like the CLOUDS phase isn't actually checked anywhere. That's okay for this PR, but I'd like a relevant TODO comment in the GameRenderer mixin as a reminder to come back to that.

**Comment by IMS212** [comment]
:+1:

**Review by coderbot16 — APPROVED** [review]
Thanks! This is a nice cleanup.

**Pull request merged**